### DISCUSSION TITLE
Get help to make contract using walletClient

### DISCUSSION THREAD

**Discussion by JaydeepAthh** [wevm/wagmi#2358] [Q&A]
I see 1.0.5 update in wagmi remove signer and provider terminology and insted use of signer now we have to use walletClient i try to make contract using my abi contract adderss and walletClient but it gives me error ( invalid signer or provider ), this is my code help to resolve this issue 

  const { data: walletClient, isError, isLoading } = useWalletClient({
    onSuccess(data) {
      console.log('Success', data)
    },
  })

  try {
    const contract = useContract({
      address: Config.CONTRACT_ADDRESS,
      abi: CONTRACT_ABI,
      walletClient,
    })
    console.log(contract)
  } catch (error) {
    console.error(error);
  }

**Comment by SaadNiaziDev** [comment]
try passing chainId to useWalletClient on which contract is deployed

**Reply by SaadNiaziDev** [reply]
went through docs and found that useContract hook is no more, its change to getContract from wagmi/actions
check this out
https://wagmi.sh/react/migration-guide#usecontract
